src/Sylius/Bundle/CoreBundle/PaymentRequest/Payum/CommandHandler/ModelPaymentRequestHandler.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\CoreBundle\PaymentRequest\Payum\CommandHandler;
+
+use Sylius\Bundle\CoreBundle\PaymentRequest\Command\PaymentRequestHashAwareInterface;
+use Sylius\Bundle\CoreBundle\PaymentRequest\Payum\Processor\RequestProcessorInterface;
+use Sylius\Bundle\CoreBundle\PaymentRequest\Payum\Provider\PaymentRequestProviderInterface;
+use Sylius\Bundle\PayumBundle\Factory\GetStatusFactoryInterface;
+use Sylius\Component\Core\Model\PaymentMethodInterface;
+use Symfony\Component\Messenger\Handler\MessageHandlerInterface;
+use Webmozart\Assert\Assert;
+
+final class ModelPaymentRequestHandler implements MessageHandlerInterface
+{
+    public function __construct(
+        private PaymentRequestProviderInterface $paymentRequestProvider,
+        private RequestProcessorInterface $requestProcessor,
+        private GetStatusFactoryInterface $factory,
+    ) {
+    }
+
+    public function __invoke(PaymentRequestHashAwareInterface $command): void
+    {
+        $paymentRequest = $this->paymentRequestProvider->provideFromHash($command->getHash());
+        Assert::notNull($paymentRequest);
```

Comment:
Please add message to assert, it will be un-debuggable otherwise.

We should make CS or arkitect rule of this for `notNull` assert. :thinking: